### PR TITLE
Addon Studies should submit both extension ID and addon URL

### PR DESCRIPTION
**Pull request by rehandalal**
r?

### CODE REVIEW

File: src/tests/workflows/recipes/components/AddonStudyFields.test.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 import { List, Map } from 'immutable';
-import OptOutStudyFields from 'console/workflows/recipes/components/OptOutStudyFields';
+import AddonStudyFields from 'console/workflows/recipes/components/AddonStudyFields';
```
Comment (mythmon) [inline src/tests/workflows/recipes/components/AddonStudyFields.test.js]: :heart_eyes:

File: src/workflows/recipes/components/RecipeForm.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ class RecipeForm extends React.PureComponent {
     'preference-experiment': PreferenceExperimentFields,
     'preference-rollout': PreferenceRolloutFields,
     'preference-rollback': PreferenceRollbackFields,
-    'opt-out-study': OptOutStudyFields,
```
Comment (mythmon) [inline src/workflows/recipes/components/RecipeForm.js]: It may be worth adding a comment here explaining the name mis-match.

**Comment by rehandalal** [comment]
> Can you add tests for this?

I tried to add tests to the `RecipeForm` component to actually test the interaction between the two fields but given the amount of wrappers and magic involved I was unable to write any.

**Comment by rehandalal** [comment]
This is blocking QA from testing addon study updates

**Pull request merged**